File: LEVEL2/문자열 압축.cpp
#include<string>
using namespace std;
 
int solution(string s) {
    int answer = s.size();
    int n = s.size() / 2;
 
    for (int i = 1; i <= n; i++) {
        string res = "";
        string tmp = s.substr(0, i);
        int cnt = 1;
 
        for (int j = i; j <= s.size(); j += i) {
            if (tmp == s.substr(j, i)) {
                cnt += 1;
            }
            
            else {
                if (cnt == 1) {
                    res += tmp;
                }
                
                else {
                    res += (to_string(cnt) + tmp);
                }
 
                tmp = s.substr(j, i);
                cnt = 1;
            }
        }
 
        if ((s.size() % i) != 0) {
            res += s.substr((s.size() / i) * i);
        }
 
        if (answer > res.size()) {
            answer = res.size();
        }
    }
 
    return answer;
}

File: LEVEL2/문자열 압축.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = s.length();
        
        if(s.length() == 1) {
            return 1;
        }
        
        for(int i = 1; i < s.length(); i++) {
            String cur = "";
            String tmp = "";
            String comp = "";
            int cnt = 1;
            
            for(int j = 0; j <= s.length() / i; j++) {
                int from = i * j;
                int to = i * (j + 1);
                
                if(to > s.length()) {
                    to = s.length();
                }
                
                cur = comp;
                comp = s.substring(from, to);
                
                if(cur.equals(comp)) {
                    cnt++;
                }
                
                else {
                    if(cnt > 1) {
                        tmp += String.valueOf(cnt);
                    }
                    
                    tmp += cur;
                    cnt = 1;
                }
            }
            
            if(cnt > 1) {
                tmp += String.valueOf(cnt);
            }
            
            tmp += comp;
            answer = Math.min(tmp.length(), answer);
        }
        
        return answer;
    }
}